lxqt-config-session/man/lxqt-config-session.1
.TH lxqt-config-session "1" "January 2025" "LXQt 2.1.0" "LXQt Configuration"
.SH NAME
\fBlxqt-config-session\fR \- Application settings of \fBLXQt\fR: The Lightweight
Qt Desktop Environment
.SH SYNOPSIS
.B lxqt-config-session
.br
.SH DESCRIPTION
This application handles \fBLXQt\fR X11 window manager selection, autostart of LXQt modules
and other applications, settings for Qt scaling, confirmation for session exit,
screenlocking, besides XDG user directories paths and advanced environment settings.
.P
If the \fBlxqt-wayland-session\fR module is installed, Wayland-related configurations
for selecting the compositor and the screenlock command will be enabled.
.SH "REPORTING BUGS"
Report bugs to https://github.com/lxqt/lxqt-session/issues
.SH "SEE ALSO"
.\" any module must refer to the session application, for module overview and initiation
\fBlxqt-session.1\fR
\fBlxqt-wayland-session.1\fR
\fBstartlxqt.1\fR
\fBstartlxqtwayland.1\fR
\fBlxqt-config.1\fR


CHANGELOG
lxqt-session-2.1.1 / 2024-12-23
================================
 * Supported compilation against procps-ng ≥ 4.0.5.

lxqt-session-2.1.0 / 2024-11-05
================================
 * Reenable lock settings under wayland.
 * Fixed the errors in KF6 check.
 * Under wayland, show the leave dialog on the screen with the cursor.
 * Don't start apps with "X-LXQt-X11-Only" under wayland.
 * Check if `wordexp` succeeds before using output (to prevent a segfault).
 * Improved the desktop entry.
 * Added a section for wayland settings.
 * Added a checkbox for "X-LXQt-X11-Only" to autostart editor.
 * Set `palette_override` to `true` by default.
 * Added tooltip for screen scaling under Wayland.
 * Added a config file for Wayland compositors.
 * Set the layer shell scope to "dialog" for the leave dialog.
 * Added GUI for custom screenlocker on x11.

lxqt-session-2.0.0 / 2024-04-17
================================
 * Made the user inputs stricter in auto-start.
 * Don't allow inputs consisting of only whitespaces, and remove start and end whitespaces from them.
 * Fixed a typo.
 * Ported to Qt6.
 * Used an overlay layer shell for the leaving dialog and centered it under Wayland.

lxqt-session-1.4.0 / 2023-11-05
================================
 * Update the DBus activation environment, to fix problems with apps (like Telegram) which set `DBusActivatable` to `true` in their desktop entries.

lxqt-session-1.3.0 / 2023-04-15
================================
 * Added support for procps-ng >= 4.0.0.
 * Use timer based checking for window manager and tray.
 * Disable crashing calls on Wayland.
 * Fixed typos.

lxqt-session-1.2.0 / 2022-11-05
================================
 * Fixed the core code when lxqt-session is started under Wayland.
 * Reload mouse & touchpad settings when devices are reconnected.

lxqt-session-1.1.1 / 2022-05-08
================================
 * Added `qtxdg-tools` as a dependency (to be used by `xdg-utils` under LXQt Session).

lxqt-session-1.1.0 / 2022-04-15
================================
 * Removed obsolete kdm session file.
 * Add NetBSD support.
 * Fixed typos in procreaper warning messages.
 * Added an option for global screen scaling.
 * Changed default icon theme to breeze.
 * Prevented cut text in header of Environment page.
 * Set Clearlooks as the default LXQt theme.

lxqt-session-1.0.1 / 2022-01-01
================================
 * Restart failed modules with exit codes other than 0.

lxqt-session-1.0.0 / 2021-11-04
================================
 * Updated AUTHORS.
 * Bumped minimum required Qt version to 5.15.

lxqt-session-0.17.1 / 2021-04-16
================================
  * Fixed compilation against Qt < 5.15.

lxqt-session-0.17.0 / 2021-04-15
================================
  * Code cleanup and modernization (use const references where possible, fix possible use of uninitialized variables, removing deprecated methods, etc.).
  * Remove Openbox configuration for WM agnosticism.
  * Completely moved to Qt5 signal/slot syntax.
  * Become subreaper and end children on finishing session.

lxqt-session-0.16.0 / 2020-11-01
================================
  * Don't use deprecated QProcess methods.
  * Don't use 0/nullptr to initialize QFlags.
  * Removed Default Applications page.

lxqt-session-0.15.0 / 2020-04-22
================================
  * Bumped version to 0.15.0.
  * C++11 code updates.
  * Removed redundant variable SAL_USE_VCLPLUGIN from startlxqt.
  * Fixed lxqt-session memory leak (triggered by wine and specific games).
  * Remove deprecated QDesktopWidget.
  * Corrected dialog location with multi-screen setups.

lxqt-session-0.14.1 / 2019-02-25
================================

  * Bumped version to 0.14.1
  * Only translations was changed.

lxqt-session-0.14.0 / 2019-01-25
================================

  * Bumped version to 0.14.0
  * Fixed implementations
    - Don't use automatic string conversions
    - confg-session: Emit each change of default app
    - lockscreenmanager: Fix renewal of inhibitor lock
    - lxqt-config-session: Fix showing translations
    - Simplify device events/actions lookup
    - Completed our WM list
    - Don't default on Openbox, make the automatic WM selection
      more reliable
  * Improved cmake scripting
    - Set cmake_minimum_required to 3.1.0
    - Removed locale compile definitions
  * Moved translations from lxqt-l10n back to lxqt-session
    - Removed obsolete translation fuctionality
    - Added translation promo in README.md
  * Translation updates
  * Cleaned up desktop file translations

lxqt-session-0.13.0 / 2018-05-21
================================

  * Spanish translation update
  * CMake: Prevent in-source builds
  * use devnode instead of devpath, so FreeBSD
  * fixed razor-qt -> lxqt and http -> https
  * Fixed some mentions of LXDE
  * Update Simplified Chinese translations (#120)
  * Use constant iterators
  * Use static QFileInfo::exists()
  * Drop Qt foreach
  * cmake: Handle CMP0063
  * install session configs in /usr/share/lxqt
  * startlxqt: Add /usr/share & /etc into XDG_CONFIG_DIRS
  * Remove left over file from the lxqt-common split
  * Add Spanish desktop entries

lxqt-session-0.12.0 / 2017-10-21
================================

  * Release 0.12.0: Update changelog
  * Set patch version
  * lxqt-session: Unify arguments parsing
  * Don't export github templates
  * Add/Update Lithuanian language
  * c'n'p bug
  * add basic cli to lxqt-session
  * lxqt-session: Provide reboot/powerOff methods
  * Drops Qt5Core_VERSION_STRING
  * lxqt-config-session: set Qt::AA_UseHighDpiPixmaps to true
  * lxqt-leave: set Qt::AA_UseHighDpiPixmaps to true
  * improved --version
  * Added QCommandLineParser and basic cli interface to lxqt-conf-session
  * Added default CMAKE_BUILD_TYPE and LXQT_SESSION_PATCH_VERSION
  * String encoding (#103)
  * Fixes a FTBFS in superbuild mode (#102)
  * lxqt-leave: Fix "disabled items" key navigation
  * Address "unused parameter" compiler warnings

lxqt-config-session/translations/lxqt-config-session.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.1">
<context>
    <name>AutoStartEdit</name>
    <message>
        <location filename="../autostartedit.ui" line="14"/>
        <source>Edit</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.ui" line="20"/>
        <source>Name:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.ui" line="40"/>
        <source>Search...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.ui" line="54"/>
        <source>Start only in X11</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.ui" line="30"/>
        <source>Command:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.ui" line="47"/>
        <source>Wait for system tray</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartedit.cpp" line="67"/>
        <source>Select Application</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>AutoStartItemModel</name>
    <message>
        <location filename="../autostartmodel.cpp" line="193"/>
        <source>Global Autostart</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartmodel.cpp" line="195"/>
        <source>LXQt Autostart</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartmodel.cpp" line="213"/>
        <source>Location: %1</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartmodel.cpp" line="215"/>
        <source>Overrides: %1</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>AutoStartPage</name>
    <message>
        <location filename="../autostartpage.ui" line="17"/>
        <source>Delete</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.ui" line="24"/>
        <source>Add</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.ui" line="31"/>
        <source>Edit</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.ui" line="44"/>
        <source>Application Autostart</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.cpp" line="140"/>
        <location filename="../autostartpage.cpp" line="151"/>
        <location filename="../autostartpage.cpp" line="170"/>
        <location filename="../autostartpage.cpp" line="187"/>
        <source>Error</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.cpp" line="140"/>
        <location filename="../autostartpage.cpp" line="170"/>
        <source>Please provide Name and Command</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../autostartpage.cpp" line="151"/>
        <location filename="../autostartpage.cpp" line="187"/>
        <source>File &apos;%1&apos; already exists!</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>BasicSettings</name>
    <message>
        <location filename="../basicsettings.ui" line="23"/>
        <source>Basic Settings</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="30"/>
        <source>Window Manager</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="49"/>
        <location filename="../basicsettings.ui" line="213"/>
        <source>Search...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="59"/>
        <source>LXQt Modules</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="65"/>
        <source>Start</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="72"/>
        <source>Stop</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="111"/>
        <source>Under Wayland, adjust scaling via compositor settings or kanshi instead.</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="114"/>
        <source>Global Screen Scaling</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="126"/>
        <source>Scale factor:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="152"/>
        <source>Leave Session</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="158"/>
        <source>Ask for confirmation to leave session</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="165"/>
        <source>Lock screen before suspending/hibernating</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="172"/>
        <source>Suspend/hibernate after lock delay:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="185"/>
        <source> ms</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.ui" line="201"/>
        <source>Use custom screen lock command</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.cpp" line="197"/>
        <source>Select a window manager</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../basicsettings.cpp" line="212"/>
        <source>Select a screenlocker</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>EnvironmentPage</name>
    <message>
        <location filename="../environmentpage.ui" line="17"/>
        <source>Add</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../environmentpage.ui" line="24"/>
        <source>Delete</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../environmentpage.ui" line="50"/>
        <source>Environment Variables (Advanced)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../environmentpage.ui" line="61"/>
        <source>Variable Name</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../environmentpage.ui" line="66"/>
        <source>Value</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>ModuleModel</name>
    <message>
        <location filename="../modulemodel.cpp" line="87"/>
        <source>Running</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>SessionConfigWindow</name>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="45"/>
        <source>LXQt Session Settings</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="48"/>
        <source>Basic Settings</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="54"/>
        <source>User Directories</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="60"/>
        <source>Autostart</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="67"/>
        <source>Wayland Settings (Experimental)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="74"/>
        <source>Environment (Advanced)</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="100"/>
        <source>Session Restart Required</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../sessionconfigwindow.cpp" line="101"/>
        <source>Some settings will not take effect until the next log in.</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>UserLocationsPage</name>
    <message>
        <location filename="../userlocationspage.cpp" line="116"/>
        <source>Locations for Personal Files</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="184"/>
        <source>LXQt Session Settings - User Directories</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="185"/>
        <source>An error occurred while applying the settings for the %1 location</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="201"/>
        <source>Choose Location</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>UserLocationsPrivate</name>
    <message>
        <location filename="../userlocationspage.cpp" line="63"/>
        <source>Desktop</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="64"/>
        <source>Downloads</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="65"/>
        <source>Templates</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="66"/>
        <source>Public Share</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="67"/>
        <source>Documents</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="68"/>
        <source>Music</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="69"/>
        <source>Pictures</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="70"/>
        <source>Videos</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="73"/>
        <source>Contains all the files which you see on your desktop</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="74"/>
        <source>Default folder to save your downloaded files</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="75"/>
        <source>Default folder to load or save templates from or to</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="76"/>
        <source>Default folder to publicly share your files</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="77"/>
        <source>Default folder to load or save documents from or to</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="78"/>
        <source>Default folder to load or save music from or to</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="79"/>
        <source>Default folder to load or save pictures from or to</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../userlocationspage.cpp" line="80"/>
        <source>Default folder to load or save videos from or to</source>
        <translation type="unfinished"></translation>
    </message>
</context>
<context>
    <name>WaylandSettings</name>
    <message>
        <location filename="../waylandsettings.ui" line="23"/>
        <source>Wayland Settings</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.ui" line="30"/>
        <source>Wayland compositor:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.ui" line="49"/>
        <location filename="../waylandsettings.ui" line="81"/>
        <source>Search...</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.ui" line="59"/>
        <source>Screenlock command:</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.ui" line="71"/>
        <source>Use &quot;loginctl lock-session&quot; on kwin_wayland</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.cpp" line="98"/>
        <source>Select a Wayland Compositor</source>
        <translation type="unfinished"></translation>
    </message>
    <message>
        <location filename="../waylandsettings.cpp" line="103"/>
        <source>Select a Screenlocker for Wayland</source>
        <translation type="unfinished"></translation>
    </message>
</context>
</TS>
